src/validation.cpp

```diff
@@ -5235,7 +5235,8 @@ class CMainCleanup
 static CMainCleanup instance_of_cmaincleanup;
 
 Optional<uint256> ChainstateManager::SnapshotBlockhash() const {
-    if (m_active_chainstate != nullptr) {
+    if (m_active_chainstate != nullptr &&
+            !m_active_chainstate->m_from_snapshot_blockhash.IsNull()) {
```

Comment:
In commit "simplify ChainstateManager::SnapshotBlockhash() return semantics" (75d1e7754274d3f015a49e71871c3a069cbe020b)

I think if all the calling code is going to continue to do `.value_or(uint256())` it makes little sense for this to return an optional. Would be simpler to just use `Optional<uint256>` everywhere (make `m_from_snapshot_blockhash` an `Optional<uint256>` and stop treating zero hash specially) or use `uint256` everywhere and keep treating zero hash specially. But having back and forth nullopt <-> null hash conversions doesn't make sense over just picking one format and sticking with it.

Reply:
Yeah I don't disagree; initially had it that way but changed it at the request of a reviewer in a previous PR.

Reply:
Cleaned up the usages to actually make use of the option. Thanks.